docs/source/usage/workflows/memoryPerDevice.py

```diff
@@ -20,72 +21,131 @@
 memory usage depends on the case and the dynamics inside the simulation.
+
 We calculate the memory of just one device per row of GPUs in laser
 propagation direction. We hereby assume that particles are distributed
 equally in transverse direction, like it is set up in the FoilLCT example.
 """
 
-
 cell_size = 0.8e-6 / 384.0  # 2.083e-9 m
 y0 = 0.5e-6  # position of foil front surface (m)
 y1 = 1.5e-6  # position of foil rear surface (m)
 L = 10e-9  # pre-plasma scale length (m)
 L_cutoff = 4 * L  # pre-plasma length (m)
 
-sim_dim = 2
-# number of cells in the simulation
-Nx_all, Ny_all, Nz_all = 256, 1280, 1
-# number of GPU rows in each direction
-x_rows, y_rows, z_rows = 2, 2, 1
-# number of cells per GPU
-Nx, Ny, Nz = Nx_all / x_rows, Ny_all / y_rows, Nz_all / z_rows
-
-vacuum_cells = ceil((y0 - L_cutoff) / cell_size)  # in front of the target
-# target cells (between surfaces + pre-plasma)
-target_cells = ceil((y1 - y0 + 2 * L_cutoff) / cell_size)
-# number of cells (y direction) on each GPU row
-GPU_rows = [0] * y_rows
-cells_to_spread = vacuum_cells + target_cells
-# spread the cells on the GPUs
-for ii, _ in enumerate(GPU_rows):
-    if cells_to_spread >= Ny:
-        GPU_rows[ii] = Ny
-        cells_to_spread -= Ny
-    else:
-        GPU_rows[ii] = cells_to_spread
-        break
-# remove vacuum cells from the front rows
-extra_cells = vacuum_cells
-for ii, _ in enumerate(GPU_rows):
-    if extra_cells >= Ny:
-        GPU_rows[ii] = 0
-        extra_cells -= Ny
+# number of vacuum cells in front of the target
+vacuum_cells = int(np.ceil((y0 - L_cutoff) / cell_size))
+# number of target cells over the depth of the target(between surfaces + pre-plasma)
+target_cells = int(np.ceil((y1 - y0 + 2 * L_cutoff) / cell_size))
+
+simulation_dimension = 2
+
+# number of cells in the simulation in each direction
+global_cell_extent = np.array((256, 1280))[:simulation_dimension]
+# number of GPUs in each direction
+global_gpu_extent = np.array((2, 2))[:simulation_dimension]
+
+super_cell_size = np.array((16, 16, 1))
+
+# how many cells each gpu in this row of this dimension handles
+#   for example grid_distribution = [np.array((64,192)), np.array((960, 320))]
+#   here we distribute cells evenly, but this is not required
+if np.any(global_cell_extent % global_gpu_extent != 0):
+    raise ValueError("global cell extent must be divisble by the global gpu extent")
+grid_distribution = []
+for dim in range(simulation_dimension):
+    row_division = np.full(global_gpu_extent[dim], int(global_cell_extent[dim] / global_gpu_extent[dim]), dtype=np.int_)
+    grid_distribution.append(row_division)
+
+print(f"grid distribution: {grid_distribution}")
+
+# get cell extent of each GPU: list of np.array[np.int_], one per simulation dimension, with each array entry being the
+#   cell extent of the corresponding gpu in the simulation, indexation by [simulation_dimension, gpu_index[0], gpu_index[1], ...]
+gpu_cell_extent = np.meshgrid(*grid_distribution)
```

Comment:
```suggestion
gpu_cell_extent = (global_cell_extent / global_gpu_extent * np.ones([simulation_dimension, *global_gpu_extent])).astype(int).T
```
Equivalent oneliner (up to the fact that we're getting a numpy array, of course). I was tempted to use an explicit `np.floor` to hint on the fact that we're potentially cutting something off here. But the `ValueError` above takes care of this. We might want to print the `gpu_cell_extent` instead of the `grid_distribution` in this case.

Reply:
as discussed offline, we decided to hard code values, for better redability.